src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractWriteDataverseCommand.java

```diff
@@ -43,42 +45,60 @@ public AbstractWriteDataverseCommand(Dataverse dataverse,
         } else {
             this.metadataBlocks = null;
         }
+        this.resetRelationsOnNullValues = resetRelationsOnNullValues;
     }
 
     @Override
     public Dataverse execute(CommandContext ctxt) throws CommandException {
         dataverse = innerExecute(ctxt);
 
+        processMetadataBlocks();
+        processFacets(ctxt);
+        processInputLevels(ctxt);
+
+        return ctxt.dataverses().save(dataverse);
+    }
+
+    private void processMetadataBlocks() {
         if (metadataBlocks != null && !metadataBlocks.isEmpty()) {
             dataverse.setMetadataBlockRoot(true);
             dataverse.setMetadataBlocks(metadataBlocks);
+        } else if (resetRelationsOnNullValues) {
+            dataverse.setMetadataBlockRoot(false);
+            dataverse.clearMetadataBlocks();
         }
+    }
 
+    private void processFacets(CommandContext ctxt) {
         if (facets != null) {
             ctxt.facets().deleteFacetsFor(dataverse);
 
             if (!facets.isEmpty()) {
                 dataverse.setFacetRoot(true);
             }
 
-            int i = 0;
-            for (DatasetFieldType df : facets) {
-                ctxt.facets().create(i++, df, dataverse);
+            for (int i = 0; i < facets.size(); i++) {
+                ctxt.facets().create(i, facets.get(i), dataverse);
             }
+        } else if (resetRelationsOnNullValues) {
+            ctxt.facets().deleteFacetsFor(dataverse);
+            dataverse.setFacetRoot(false);
         }
+    }
 
+    private void processInputLevels(CommandContext ctxt) {
         if (inputLevels != null) {
             if (!inputLevels.isEmpty()) {
                 dataverse.addInputLevelsMetadataBlocksIfNotPresent(inputLevels);
             }
             ctxt.fieldTypeInputLevels().deleteFacetsFor(dataverse);
-            for (DataverseFieldTypeInputLevel inputLevel : inputLevels) {
+            inputLevels.forEach(inputLevel -> {
                 inputLevel.setDataverse(dataverse);
                 ctxt.fieldTypeInputLevels().create(inputLevel);
-            }
+            });
+        } else if (resetRelationsOnNullValues) {
+            ctxt.fieldTypeInputLevels().deleteFacetsFor(dataverse);
```

Comment:
Same question - does this belong in process input levels?

Reply:
Answered in the comment above!